### PR TITLE
Ping rpc and grpc connections

### LINKED ISSUE #1: RPC: Ping
To test whether a node is online

### PR DESCRIPTION
**Pull request by Kyrne**
Setup grpc connections for server and client
Defined the protobuf for ping and implemented it

Added a grpc cli to debug/test rpcs in docker containers

Added ci for test, test with race flag and build

closes #1

### CODE REVIEW

File: .github/workflows/test.yml
Comment (maxluetkemeyer) [inline .github/workflows/test.yml:1]: The workflow environment and the docker base image should be the same. Is it?

File: Dockerfile
Comment (maxluetkemeyer) [inline Dockerfile:1]: we should define the port here as env variable

File: kademlia/bucket/add_contact.go
Comment (maxluetkemeyer) [inline kademlia/bucket/add_contact.go:1]: If we do it like this, the extra function "AddContact" is not necessary anymore and we can rename the custom func to "AddContact"

**Review by maxluetkemeyer — CHANGES_REQUESTED** [review]
I propose to construct the network stuff over interfaces. It is easier to test then and we can have a gRpc, local and testing network then.

**Review by hugocornago-ltu — COMMENTED** [review]
I don't think there should be a "client" and a "server" in our implementation.
We should only care about the servers communicating each other, not about the client.

File: docker-compose.yml
[the remark this answers is not in the document]
Comment (Kyrne) [reply docker-compose.yml:1]: Did not mean to push this, will revert back to 50.

**Pull request merged**